File: Linguagem C#/diagonal_negativos.cs
// See https://aka.ms/new-console-template for more information
using System.Globalization;
using System.Runtime.CompilerServices;
using System.Runtime.Serialization;
CultureInfo CI = CultureInfo.InvariantCulture;

Console.Write("Qual a ordem da matriz? ");
int n = int.Parse(Console.ReadLine());

int[,] mat = new int[n,n];
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        Console.Write("Elemento ["+i+","+j+"]:");
        mat[i,j] = int.Parse(Console.ReadLine());
    }
}

Console.Write("DIAGONAL PRINCIPAL: ");
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        if (i == j) {
            Console.Write(mat[i,j] + " ");
        }
    }
}

int soma = 0;  
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
       if (mat[i,j] < 0) {
            soma++;
        }
    }
}
Console.WriteLine();
Console.Write("QUANTIDADE DE NEGATIVOS = " + soma);

File: Linguagem C++/diagonal_negativos.cpp
#include <iostream>

using namespace std;

int main() {

    int n,i,j,negativos;
    cout << "Qual a ordem da matriz? ";
    cin >> n;
    int mat[n][n];

    for(i = 0; i < n; i++) {
        for(j = 0; j < n; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }
    cout << "DIAGONAL PRINCIPAL: \n";
    for(i = 0; i < n; i++) {
        for(j = 0; j < n; j++) {
            if(i == j) {
                cout << " " << mat[i][j];
            }
        }
    }

     negativos = 0;
     for(i = 0; i < n; i++) {
        for(j = 0; j < n; j++) {
            if (mat[i][j] < 0) {
                negativos++;
            }
        }
    }
    cout << "\n";
    cout << "QUANTIDADE DE NEGATIVOS = " << negativos;


    return 0;
}
